src/berry_mill/mill.py

```diff
@@ -81,6 +81,10 @@ def run(self) -> None:
         Build an image
         """
 
+        # if target_dir exists with no --clean option, fail early.
+        if not self.args.clean and os.path.isdir(self.args.target_dir):
```

Comment:
Could be also beneficial to check if parent is writable, in case target does not exists. Because it will still fail if a user sets it to some read-only area.

Reply:
Done.